pandas/core/window.py

```diff
@@ -671,14 +679,24 @@ def _apply_window(self, mean=True, **kwargs):
         window = self._prep_window(**kwargs)
         center = self.center
 
-        blocks, obj, index = self._create_blocks()
+        blocks_dict, obj, index = self._create_blocks()
+        dtypes = blocks_dict.keys()
+        blocks = blocks_dict.values()
```

Comment:
do you need blocks? (as you are using blocks_dict)

Reply:
It is actually used when calling `_wrap_results` [here](https://github.com/pandas-dev/pandas/blob/ad9b5e2aeaaa32ced72ef2be65d7067c3bce0345/pandas/core/window.py#L721). I guess it is better to relocate it just above return.